autolens/point/analysis.py

```diff
@@ -1,20 +1,21 @@
-from typing import List
+import math
+from abc import ABC, abstractmethod
+from typing import List, Tuple
 
 import autofit as af
 from autoarray import Grid2D
-from autogalaxy.profiles.point_sources import Point
 from autolens.point.triangles.triangle_solver import TriangleSolver
 
 
-class Analysis(af.Analysis):
+class PointSourceAnalysis(af.Analysis, ABC):
     def __init__(
```

Comment:
Generally use convention of `dataset` goes into `Analysis` and it makes `model_data`.

This is a bit geneirc, but unifies the API with other `Analysis` objects.

Reply:
Seems like overkill at the moment but I may update that as this develops